cross/toolchain.cmake

```diff
@@ -72,7 +73,9 @@ elseif(TARGET_ARCH_NAME MATCHES "^(arm|arm64)$")
   add_compile_param(CROSS_LINK_FLAGS "-L${CROSS_ROOTFS}/lib/${TOOLCHAIN}")
 else() # x86 case
   add_compile_param(CROSS_LINK_FLAGS "-B${CROSS_ROOTFS}/usr/lib/gcc/i686-linux-gnu")
-  add_compile_param(CROSS_LINK_FLAGS "-L${CROSS_ROOTFS}/lib/i386-linux-gnu")
+  add_compile_param(CROSS_LINK_FLAGS "-L${CROSS_ROOTFS}/lib/${TOOLCHAIN}")
+  add_compile_param(CROSS_LINK_FLAGS "-L${CROSS_ROOTFS}/usr/lib/${TOOLCHAIN}")
+  add_compile_param(CROSS_LINK_FLAGS "-Wl,-rpath-link,${CROSS_ROOTFS}/usr/lib/${TOOLCHAIN}")
```

Comment:
Is there a reason why this one has `rpath-link`, but none of the other ones?

Reply:
Yes, the compiler itself suggests using the rpath-link. Other methods as for other architectures using -B and -L don't work in this case. Another working method is to specify when linking "coreclrtraceptprovider" other libraries directly "-lurcu-bp -lurcu-cds -llttng-ust-tracepoint". But I like this method less. It's just the dependencies that the shared library lttng-ust already has.
...
Now - I understand the source of all troubles.
The problem is that old bfd linker is used on x86. I'll fix it another way with forcing gold.
This will exclude possible similar errors for other architectures too.